tests/lib/data/alignment-group.test.js

```diff
@@ -449,13 +460,373 @@ describe('alignment-group.test.js', () => {
+
+    alGroup1.add(token3)
+    expect(alGroup1.groupLen).toEqual(2)
+    expect(alGroup1.steps.length).toEqual(2)
+
+    expect(alGroup1.includesToken(token1)).toBeTruthy()
+    expect(alGroup1.includesToken(token2)).toBeFalsy()
+    expect(alGroup1.includesToken(token3)).toBeTruthy()
+
+    expect(alGroup1.steps[0]).toHaveProperty('token', token1)
+    expect(alGroup1.steps[0]).toHaveProperty('type', 'add')
+
+    expect(alGroup1.steps[1]).toHaveProperty('token', token3)
+    expect(alGroup1.steps[1]).toHaveProperty('type', 'add')
+  })
+
+  it('28 AlignmentGroup - remove method - would remove undone steps, if we add a new token to a group', () => {
+    const alGroup1 = new AlignmentGroup()
+
+    const token1 = new Token({
+      textType: 'origin', idWord: 'L1-10', word: 'male'})
+
+    const token2 = new Token({
+      textType: 'target', idWord: 'L2-10', word: 'man'})
+
+    const token3 = new Token({
+      textType: 'target', idWord: 'L2-11', word: 'woman'})
+
+    alGroup1.add(token1)
+    alGroup1.add(token2)
+    alGroup1.add(token3)
+    expect(alGroup1.groupLen).toEqual(3)
+    expect(alGroup1.steps.length).toEqual(3)
+
+    alGroup1.undo()
+
+    expect(alGroup1.groupLen).toEqual(2)
+    expect(alGroup1.steps.length).toEqual(3)
+
+    alGroup1.remove(token2)
+
+    expect(alGroup1.groupLen).toEqual(1)
+    expect(alGroup1.steps.length).toEqual(3)
+
+    expect(alGroup1.includesToken(token1)).toBeTruthy()
+    expect(alGroup1.includesToken(token2)).toBeFalsy()
+    expect(alGroup1.includesToken(token3)).toBeFalsy()
+
+    expect(alGroup1.steps[0]).toHaveProperty('token', token1)
+    expect(alGroup1.steps[0]).toHaveProperty('type', 'add')
+
+    expect(alGroup1.steps[1]).toHaveProperty('token', token2)
+    expect(alGroup1.steps[1]).toHaveProperty('type', 'add')
+
+    expect(alGroup1.steps[2]).toHaveProperty('token', token2)
+    expect(alGroup1.steps[2]).toHaveProperty('type', 'remove')
+  })
+
+  it('24 AlignmentGroup - unmerge removes tokens from the group and returns previously merged group as a result', () => {
+    const alGroup1 = new AlignmentGroup()
+
+    const token1 = new Token({
+      textType: 'origin', idWord: 'L1-10', word: 'male'})
+
+    const token2 = new Token({
+      textType: 'target', idWord: 'L2-10', word: 'man'})
+
+    alGroup1.add(token1)
+    alGroup1.add(token2)
+
+    const alGroup2 = new AlignmentGroup()
+
+    const token3 = new Token({
+      textType: 'origin', idWord: 'L1-8', word: 'mare'})
+
+    const token4 = new Token({
+      textType: 'target', idWord: 'L2-8', word: 'sea'})
+
+    alGroup2.add(token3)
+    alGroup2.add(token4)
+
+    alGroup1.merge(alGroup2, 1)
+
+    const result = alGroup1.unmerge(alGroup1.steps[alGroup1.steps.length - 1])
+
+    expect(alGroup1.includesToken(token1)).toBeTruthy()
+    expect(alGroup1.includesToken(token2)).toBeTruthy()
+    expect(alGroup1.includesToken(token3)).toBeFalsy()
+    expect(alGroup1.includesToken(token4)).toBeFalsy()

```

Comment:
maybe it would be good to test the tokens prior to the unmerge step too, just to make sure the unmerge is actually removing something?

Reply:
ok - will add